topo/node/node.go

```diff
@@ -220,6 +228,79 @@ func (n *Impl) Create(ctx context.Context) error {
 	return nil
 }
 
+// ValidateConstraints will check the host constraints and returns a list of errors for cases which do not meet
+// the constraints
+func (n *Impl) ValidateConstraints() error {
+	hostConstraints := n.GetProto().GetHostConstraints()
+	if hostConstraints == nil {
+		return nil
+	}
+	var errorList errlist.List
+	for _, hc := range hostConstraints {
+		switch v := hc.GetConstraint().(type) {
+		case *tpb.HostConstraint_KernelConstraint:
+			log.Infof("Validating %s constraint for node %s", v.KernelConstraint.GetName(), n.String())
+			kcValue, err := kernelConstraintValue(n.BasePath, v.KernelConstraint.Name)
+			if err != nil {
+				errorList.Add(err)
+				continue
+			}
+			switch k := v.KernelConstraint.GetConstraintType().(type) {
+			case *tpb.KernelParam_BoundedInteger:
+				if err := validateBoundedInteger(k.BoundedInteger, kcValue); err != nil {
+					errorList.Add(fmt.Errorf("failed to validate kernel constraint error: %w", err))
+				}
+			}
+		default:
+			return nil
+		}
+	}
+
+	return errorList.Err()
+}
+
+var (
+	kernelConstraintValue = func(basePath string, constraint string) (int, error) {
```

Comment:
why does this require a base path? Aren't all of the sysctl values found under the root dir?

Reply:
My bad, had added it initially for testing. Removed it.